src/metabase/driver/clickhouse.clj

```diff
@@ -294,15 +298,15 @@
 ;; because it eats some performance, but I do not know a better solution
 (defmethod sql.qp/->honeysql [:clickhouse ZonedDateTime]
   [_ t]
-  (hsql/call :parseDateTimeBestEffort t))
+  (hsql/call :parseDateTime64BestEffort (t/format "yyyy-MM-dd HH:mm:ss.SSSZZZZZ" t)))
```

Comment:
is it necessary to check the precision first and then decide to use `parseDateTimeBestEffort` or `parseDateTime64BestEffort`?

Reply:
@lucas-tubi, I'd say yes

```
SELECT count()
FROM system.numbers
WHERE NOT ignore(toDateTime(rand()))

↘ Progress: 9.94 billion rows, 79.50 GB (2.42 billion rows/s., 19.35 GB/s.)                                                                                                (1.0 CPU)

Query was cancelled.

0 rows in set. Elapsed: 4.309 sec. Processed 10.42 billion rows, 83.39 GB (2.42 billion rows/s., 19.35 GB/s.)



SELECT count()
FROM system.numbers
WHERE NOT ignore(toDateTime64(rand(), 0))

← Progress: 2.48 billion rows, 19.82 GB (475.59 million rows/s., 3.80 GB/s.)                                                                                               (1.0 CPU)

Query was cancelled.

0 rows in set. Elapsed: 5.410 sec. Processed 2.57 billion rows, 20.58 GB (475.57 million rows/s., 3.80 GB/s.)
```